core/src/main/java/org/apache/accumulo/core/client/security/tokens/KerberosToken.java

```diff
@@ -51,11 +52,11 @@
    *          The user that is logged in
    */
   public KerberosToken(String principal) throws IOException {
-    requireNonNull(principal);
+    this.principal = requireNonNull(principal);
     final UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
-    checkArgument(ugi.hasKerberosCredentials(), "Subject is not logged in via Kerberos");
-    checkArgument(principal.equals(ugi.getUserName()), "Provided principal does not match currently logged-in user");
-    this.principal = ugi.getUserName();
+    if (AuthenticationMethod.KERBEROS == ugi.getAuthenticationMethod()) {
+      checkArgument(ugi.hasKerberosCredentials(), "Subject is not logged in via Kerberos");
+    }
```

Comment:
Drat, I pushed without re-checking my email.

I mulled around whether or not we should even have this check (I haven't traced the hadoop code to understand whether or not this is even possible as-written). I probably would have just suggested to remove it entirely.

Having it check for PROXY or KERBEROS might be a good middle-ground (failing on SIMPLE, TOKEN, etc). I can spin out a follow-on issue since I already pushed this and it was only tweaking the original, already sub-par logic as you said.